main/src/ca/uwaterloo/flix/language/phase/jvm/ClassMaker.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020-2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.jvm
+
+import ca.uwaterloo.flix.api.Flix
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.Opcodes._
+
+class ClassMaker(visitor: ClassWriter) {
+  private def makeField(fieldName: String, fieldType: JvmType, mod: Int): Unit = {
+    val field = visitor.visitField(mod, fieldName, fieldType.toDescriptor, null, null)
+    field.visitEnd()
+  }
+
+  def mkField(fieldName: String, fieldType: JvmType): Unit = {
```

Comment:
I would add two optional arguments with some new enums: Visibility: Public/Private and Finality: Final/NonFinal.

Reply:
And then just ignore the third option of neither public or private? I was unsure of that so I held back on it

Reply:
There is protected and package protected, but I dont think we will use those.